extensions/amp-ad/0.1/amp-ad-custom.js

```diff
@@ -85,7 +85,7 @@ export class AmpAdCustom extends AMP.BaseElement {
     /** @const {string} fullUrl */
     const fullUrl = this.getFullUrl_();
     // If this promise has no URL yet, create one for it.
-    if (!(fullUrl in ampCustomadXhrPromises)) {
+    if (this.slot_ === null || !(fullUrl in ampCustomadXhrPromises)) {
```

Comment:
if two requests share the same fullUrl, Wouldn't `ampCustomadXhrPromises[fullUrl]` be replaced by the second request.

Reply:
Isn't that what we would want if there is no `data-slot`?

Reply:
Well functional wise they are the same thing. But we don't really need to check with `ampCustomadXhrPromises` array if there's no `data-slot`. What's the point of keeping them in the array.

Reply:
To clarify, we don't want to keep something we won't used later to the `ampCustomadXhrPromises` map. Can we do something like
```
let promise = null;
promise = ampCustomadXhrPromises[fullUrl] || Services.xhrFor(this.win)....
if (this.slot_) {
  ampCustomadXhrPromises[fullUrl] = promise;
}
return promise.then(() => {});
```

Reply:
You are right, this is better. Refactored.